packages/core/src/codewhispererChat/tools/chatStream.ts

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Writable } from 'stream'
+import { getLogger } from '../../shared/logger/logger'
+import { Messenger } from '../controllers/chat/messenger/messenger'
+
+/**
+ * A writable stream that feeds each chunk/line to the chat UI.
+ * Used for streaming tool output (like bash execution) to the chat interface.
+ */
+export class ChatStream extends Writable {
+    private accumulatedLogs = ''
+
+    public constructor(
+        private readonly messenger: Messenger,
+        private readonly tabID: string,
+        private readonly triggerID: string,
+        private readonly logger = getLogger('chatStream')
+    ) {
+        super()
+        this.logger.debug(`ChatStream created for tabID: ${tabID}, triggerID: ${triggerID}`)
+    }
+
+    override _write(chunk: Buffer, encoding: BufferEncoding, callback: (error?: Error | null) => void): void {
+        const text = chunk.toString()
+        this.accumulatedLogs += text
+        this.logger.debug(`ChatStream received chunk: ${text}`)
+        this.messenger.sendPartialBashToolLog(`\`\`\`bash\n${this.accumulatedLogs}\`\`\``, this.tabID, this.triggerID)
```

Comment:
Is there a way we can make this generic and not mention `bash`? I'm wondering if this can be reused by the other tools

Reply:
The method can be renamed and used by other tools. I added `bash` because there seems to be some processing on the chat message field where for text in triple-backticks (```) it tries to extract the first bit of text after the backticks to be used as the language name for the code block.

Reply:
Yep the UI uses the markdown format to determine the code block language. But I'm thinking for example, if we want to have a tool print some other code block language then it shouldn't be hardcoded as bash here. But let's follow up later